assets/src/scripts/monitoring-location/components/hydrograph/index.test.js

```diff
@@ -217,390 +83,220 @@ describe('monitoring-location/components/hydrograph module', () => {
+            expect(retrieveWaterwatchDataSpy).toHaveBeenCalled();
+            expect(retrieveHydrographParametersSpy).not.toHaveBeenCalled();
         });
     });
 
-    describe('graphNode contains the expected elements when no IV time series has been retrieved and showOnlyGraph is false', () => {
+    describe('Tests for rendering once fetching is complete when showOnlyGraph is false', () => {
         let store;
-        config.NWIS_INVENTORY_PAGE_URL = 'https://fakenwis.usgs.gov/inventory';
-        let resolvedLoadPromise = Promise.resolve();
         beforeEach(() => {
-            jest.spyOn(floodDataActions, 'retrieveWaterwatchData').mockReturnValue(function() {
-                return Promise.resolve({});
-            });
-            jest.spyOn(ivTimeSeriesDataActions, 'retrieveIVTimeSeries').mockReturnValue(function() {
-                return Promise.resolve({});
-            });
             store = configureStore({
-                ...TEST_STATE,
-                ivTimeSeriesData: {},
-                ivTimeSeriesState: {
-                    ...TEST_STATE.ivTimeSeriesState,
-                    currentIVVariableID: '',
-                    currentIVDateRange: ''
-
+                hydrographData: {
+                    primaryIVData: TEST_PRIMARY_IV_DATA,
+                    currentTimeRange: TEST_CURRENT_TIME_RANGE,
+                    groundwaterLevels: TEST_GW_LEVELS,
+                    medianStatisticsData: TEST_MEDIAN_DATA
                 },
+                hydrographState: {
+                    selectedIVMethodID: '90649',
+                    showMedianData: true
+                },
+                hydrographParameters: TEST_HYDROGRAPH_PARAMETERS,
+                floodData: {},
                 ui: {
-                    windowWidth: 400,
-                    width: 400
+                    width: 1000
                 }
             });
-            attachToNode(store, graphNode, {siteno: '12345678'}, resolvedLoadPromise);
-            return new Promise(resolve => {
-                window.requestAnimationFrame(() => {
-                    resolve();
-                });
-            });
-        });
-
-        it('should show info alert', () => {
-            expect(select('.usa-alert--info').size()).toBe(1);
-        });
 
-        it('should use inventory for classic page link', () => {
-            expect(select('#classic-page-link').attr('href')).toContain('https://fakenwis.usgs.gov/inventory');
-        });
-    });
-
-    describe('graphNode contains the expected elements when showOnlyGraph is false', () => {
-        /* eslint no-use-before-define: 0 */
-        let store;
-        let resolvedLoadPromise = Promise.resolve();
-        beforeEach(() => {
-            jest.spyOn(floodDataActions, 'retrieveWaterwatchData').mockReturnValue(function() {
-                return Promise.resolve({});
+            hydrographData.retrieveHydrographData = jest.fn(() => {
```

Comment:
This is the function I was having trouble mocking. The asynchronous actions function return a function which returns a function that returns a promise. I just needed to really look at what the real asynchronous actions did.